packages/patrol_cli/lib/src/commands/build_ios.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:path/path.dart' show basename;
+import 'package:patrol_cli/src/base/exceptions.dart';
+import 'package:patrol_cli/src/base/extensions/core.dart';
+import 'package:patrol_cli/src/base/logger.dart';
+import 'package:patrol_cli/src/crossplatform/app_options.dart';
+import 'package:patrol_cli/src/dart_defines_reader.dart';
+import 'package:patrol_cli/src/ios/ios_test_backend.dart';
+import 'package:patrol_cli/src/pubspec_reader.dart';
+import 'package:patrol_cli/src/runner/patrol_command.dart';
+import 'package:patrol_cli/src/test_finder.dart';
+
+class BuildIOSCommand extends PatrolCommand {
+  BuildIOSCommand({
+    required TestFinder testFinder,
+    required DartDefinesReader dartDefinesReader,
+    required PubspecReader pubspecReader,
+    required IOSTestBackend iosTestBackend,
+    required Logger logger,
+  })  : _testFinder = testFinder,
+        _dartDefinesReader = dartDefinesReader,
+        _pubspecReader = pubspecReader,
+        _iosTestBackend = iosTestBackend,
+        _logger = logger {
+    usesTargetOption();
+    usesBuildModeOption();
+    usesFlavorOption();
+    usesDartDefineOption();
+    usesLabelOption();
+    usesWaitOption();
+
+    usesIOSOptions();
+    argParser.addFlag(
+      'simulator',
+      help: 'Build for simulator instead of real device.',
+    );
+  }
+
+  final TestFinder _testFinder;
+  final DartDefinesReader _dartDefinesReader;
+  final PubspecReader _pubspecReader;
+  final IOSTestBackend _iosTestBackend;
+
+  final Logger _logger;
+
+  @override
+  String get name => 'ios';
+
+  @override
+  String get description => 'Build app for integration testing on iOS.';
+
+  @override
+  Future<int> run() async {
+    final targetArg = stringsArg('target');
+    if (targetArg.isEmpty) {
+      throwToolExit('No test target specified');
+    } else if (targetArg.length > 1) {
+      throwToolExit('Only one test target can be specified');
+    }
+    final target = _testFinder.findTest(targetArg.single);
+    _logger.detail('Received test target: $target');
+
+    final config = _pubspecReader.read();
+    final flavor = stringArg('flavor') ?? config.ios.flavor;
+    if (flavor != null) {
+      _logger.detail('Received iOS flavor: $flavor');
+    }
+
+    final bundleId = stringArg('bundle-id') ?? config.ios.bundleId;
```

Comment:
From that point on, to after `flutterOpts` definition there is a common code for Android and iOS (and possible future platforms as I suppose). Can this be unified and not repeated?

I'm not talking about plain simple code repetition, but more about the context, the context of this code (looks to me) is the same.

Reply:
Yup, I see.

In general, I had a few approaches to introducing some more rigid abstractions in `patrol_cli`, but in the end, the benefit was minimal and it was only constraining.

I'm going to invoke `flutter_tools` as well - from my reading of their code, they prefer to duplicate code rather than to write a (almost always) leaky abstraction to "unify and not repeat". There are just too many subtle differences between Android and iOS build&test processes (and more in the future).

I would _love_ for the codebase to better, but I think it's just not a problem at our scale. And apparently, at `flutter_tools`'s scale as well.